lib/internal/mime.js

```diff
@@ -0,0 +1,279 @@
+const ENDING_WHITESPACE = new RegExp(`[${HTTP_WHITESPACE.replace(/\W/g, '\\$&')}]*$`);
+function toASCIILower(str) {
+  return StringPrototypeReplace(str, /[A-Z]/g, c => StringPrototypeToLowerCase(c))
+}
+function parse(str, target) {
+  str = StringPrototypeReplace(str, BEGINNING_WHITESPACE, '');
+  str = StringPrototypeReplace(str, ENDING_WHITESPACE, '');
+  let position = 0;
+  let type = exec(/^[^\/]*/, str);
+  if (!type || type[0] === '' || test(NotHTTPTokenCodePoint, type[0])) {
+    throw new TypeError('Error parsing MIME: invalid type');
+  }
+  type = type[0];
+  position = type.length;
+  position += 1;
+  let subtype = exec(/^[^;]*/, StringPrototypeSlice(str, position));
+  if (subtype) {
+    position = position += subtype[0].length;
+    subtype[0] = StringPrototypeReplace(subtype[0], ENDING_WHITESPACE, '');
+  }
+  if (!subtype || subtype[0] === '' ||test(NotHTTPTokenCodePoint, subtype[0])) {
+    throw new TypeError('Error parsing MIME: invalid subtype');
+  }
+  subtype = subtype[0];
+  target.type = toASCIILower(type);
+  target.subtype = toASCIILower(subtype);
+  const parameters = { __proto__: null };
+  while (position < str.length) {
+    position++;
+    const skip = exec(BEGINNING_WHITESPACE, StringPrototypeSlice(str, position));
+    if (skip) position += skip[0].length;
+    let parameterName = exec(/^[^;=]*/, StringPrototypeSlice(str, position));
+    parameterName = parameterName[0];
+    position += parameterName.length;
+    parameterName = toASCIILower(parameterName);
+    if (position < str.length) {
+      const [codePoint] = StringPrototypeSlice(str, position);
+      if (codePoint === ';') {
+        continue;
+      } else {
+        position += 1;
+      }
+    }
+    if (position >= str.length) {
+      break;
+    }
+    const [codePoint] = StringPrototypeSlice(str, position);
+    let parameterValue = null;
+    if (codePoint === '"') {
+      position += 1;
+      const inside = exec(/^(?:([\\]$)|[\\][\s\S]|[^"])*(?:(")|$)/u, StringPrototypeSlice(str, position));
+      position += inside[0].length;
+      // Trailing \ at end of string
+      parameterValue = StringPrototypeReplace(StringPrototypeSlice(
+        inside[0],
+        0,
+        inside[1] || inside[2] ? -1 : undefined
+      ), /[\\]([\s\S])/ug, '$1');
+      if (inside[1]) parameterValue += '\\';
+    } else {
+      let value = exec(/[^;]*/, StringPrototypeSlice(str, position))[0];
+      position += value.length;
+      value = StringPrototypeReplace(value, ENDING_WHITESPACE, '');
+      if (value === '') continue;
+      parameterValue = value;
+    }
+    if (parameterName !== '' &&
+      test(NotHTTPTokenCodePoint, parameterName) === false &&
+      test(NotHTTPQuotedStringCodePoint, parameterValue) === false &&
+      parameterName in parameters === false) {
+      parameters[parameterName] = parameterValue;
+    }
+  }
+  target.parameters = ObjectEntries(parameters);
+  return target;
+}
```

Comment:
This function is very dense, could you add some comments and blank lines?

Reply:
i've added some comments and done a split for type/subtype and params